### ISSUE TITLE
4.0.6 not opening plot windows

### ISSUE THREAD

**Issue by jamesdorsey** [blink1073/oct2py#129]
Tried to use 4.0.6 with Python 2.7.13 on Ubuntu.

`from oct2py import octave as oct`
`oct.plot(1,1)`
`<some negative float>`

But no plot window appears. Works fine after downgrade to 4.0.0. Let me know if you need further info to replicate.

**Comment by blink1073** [comment]
The culprit is this [commit](https://github.com/blink1073/oct2py/commit/01b85951c6c5cd4f6e9c580a8d8e1584f17399e6).  I need to figure out a better way to handle the graphics toolkit.

**Comment by jamesdorsey** [comment]
Might it be possible to invoke the octave session as

`octave --no-gui`

Instead of

`octave-cli`

And then use the qt toolkit? I think that would be my ideal solution, although I've probably missed something massively obvious...

**Comment by blink1073** [comment]
That might work, but I remember there being a conflict with event loops.

**Comment by kupiqu** [comment]
> The culprit is this [commit](https://github.com/blink1073/oct2py/commit/01b85951c6c5cd4f6e9c580a8d8e1584f17399e6). I need to figure out a better way to handle the graphics toolkit.

Current official graphics toolkit is qt, I guess it doesn't work in oct2py, does it?

**Comment by kupiqu** [comment]
There is a warning:

`__main__:1: UserWarning: Key - value pairs are deprecated, use 'func_args'`

But it actually works!

**Comment by kupiqu** [comment]
Actually, may I ask why did you introduce the warning? There are many arguments in Octave such as linewidth, so it's going to be very hard to add them all to the `allowed` list of arguments in dynamic.py. Or am I missing sth?

**Comment by blink1073** [comment]
Hi @kupiqu, I don't follow, which code did you change?

**Comment by kupiqu** [comment]
for allowing figures to appear, just replace None by 'qt' in the commit above:

```
settings = dict(backend='inline' if plot_dir else 'qt',
￼	                        format=kwargs.get('plot_format'),
￼	                        name=kwargs.get('plot_name'),
￼	                        width=kwargs.get('plot_width'),
```

the warning appears when invoking plot with a linewidth specification (taken from your demo). I commented out this in dynamic.py:

```
        # if extras:
        #     warnings.warn('Key - value pairs are deprecated, use `func_args`',
        #                   stacklevel=2)
```

**Comment by blink1073** [comment]
I released 4.1.0 which allows you to specify a backend property.  I also fixed that demo.  We don't allow kwargs to be used because we had some reserved keyword arguments.

**Comment by kupiqu** [comment]
Excellent
